datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1074,6 +1074,13 @@ impl Projection {
             alias,
         })
     }
+
+    pub fn try_from_plan(plan: &LogicalPlan) -> datafusion_common::Result<&Projection> {
```

Comment:
For what it is worth, I think a more rust idiomatic way would be to `impl std::convert::TryInto`

https://doc.rust-lang.org/std/convert/trait.TryInto.html

However, I think this way is also fine

Reply:
I started with that, but ran into borrow checker / trait issues with returning a reference, vs cloning the underlying data (I'm sure I could have gotten it to work with a clone). I was thinking that maybe this method should be called `try_from_plan_ref` in order to differentiate it from any future non-ref version. 50% of this PR was just trying to de-verbosify those plan re-writers into something that was readable and fit on a page.

Reply:
I think it is fine as is